Ladybird/CMakeLists.txt

```diff
@@ -47,6 +47,13 @@ if (ENABLE_UNDEFINED_SANITIZER)
     add_link_options(-fsanitize=undefined)
 endif()
 
+if (HAIKU)
+    # Haiku needs some extra compile definitions to make important stuff in its headers available.
+    add_compile_definitions(_DEFAULT_SOURCE)
+    add_compile_definitions(_GNU_SOURCE)
+    add_compile_definitions(__USE_GNU)
```

Comment:
These actually shouldn't be needed. The headers will automatically define them for you so long as neither `__STRICT_ANSI__` nor `_POSIX_C_SOURCE` is defined. By default, GCC defines one or both of those when `-std=c++...` or `-std=c...` is specified; using `-std=gnu++...` or `-std=gnu...` will resolve that problem and avoid this.

Reply:
Are you sure about that?
I tried changing CMAKE_CXX_EXTENSIONS to ON and it does now use -std=gnu++20 but the build now fails because it doesn't find sighandler_t anymore.
```
~/serenity2> cmake --build Build/ladybird -j1
[0/2] Re-checking globbed directories...
[1/2] Re-running CMake...
-- Configuring done
-- Generating done
-- Build files have been written to: /boot/home/serenity2/Build/ladybird
[0/2] Re-checking globbed directories...
[1/2328] Building CXX object Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/AnonymousBuffer.cpp.o
FAILED: Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/AnonymousBuffer.cpp.o 
/bin/c++ -DENABLE_COMPILETIME_FORMAT_CHECK -DLibCore_EXPORTS -I/boot/home/serenity2 -I/boot/home/serenity2/Userland/Services -I/boot/home/serenity2/Userland/Libraries -I/boot/home/serenity2/Build/ladybird/Lagom -I/boot/home/serenity2/Build/ladybird/Lagom/Userland/Services -I/boot/home/serenity2/Build/ladybird/Lagom/Userland/Libraries -I/boot/home/serenity2/Meta/Lagom/../.. -I/boot/home/serenity2/Meta/Lagom/../../Userland -I/boot/home/serenity2/Meta/Lagom/../../Userland/Libraries -I/boot/home/serenity2/Meta/Lagom/../../Userland/Services -I/boot/home/serenity2/Build/ladybird -fPIC -Wall -Wextra -Wno-unknown-warning-option -Wno-unused-command-line-argument -fdiagnostics-color=always -fno-exceptions -Werror -Wno-expansion-to-defined -Wno-literal-suffix -Wno-dangling-reference -fno-semantic-interposition -fvisibility-inlines-hidden -Wno-maybe-uninitialized -Wno-shorten-64-to-32 -fsigned-char -g1 -O2 -fPIC -std=gnu++20 -MD -MT Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/AnonymousBuffer.cpp.o -MF Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/AnonymousBuffer.cpp.o.d -o Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/AnonymousBuffer.cpp.o -c /boot/home/serenity2/Userland/Libraries/LibCore/AnonymousBuffer.cpp
In file included from /boot/home/serenity2/Userland/Libraries/LibCore/AnonymousBuffer.cpp:9:
/boot/home/serenity2/Userland/Libraries/LibCore/System.h:118:9: error: 'sighandler_t' was not declared in this scope
  118 | ErrorOr<sighandler_t> signal(int signal, sighandler_t handler);
      |         ^~~~~~~~~~~~
/boot/home/serenity2/Userland/Libraries/LibCore/System.h:118:21: error: template argument 1 is invalid
  118 | ErrorOr<sighandler_t> signal(int signal, sighandler_t handler);
      |                     ^
/boot/home/serenity2/Userland/Libraries/LibCore/System.h:118:42: error: 'sighandler_t' has not been declared
  118 | ErrorOr<sighandler_t> signal(int signal, sighandler_t handler);
      |                                          ^~~~~~~~~~~~
cc1plus: note: unrecognized command-line option '-Wno-shorten-64-to-32' may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option '-Wno-unused-command-line-argument' may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
ninja: build stopped: subcommand failed.
```

Reply:
That's behind `__USE_GNU` even on glibc, it appears? Hmm, we may want to fix that for Haiku anyway.

Reply:
Done in hrev57250.

Reply:
I just retried building, now running on hrev57250.
The sighandler_t is fine now, but pthread_getattr_np, which is hidden behind a _GNU_SOURCE ifdef, is still missing:
```
~/serenity2> cmake --build Build/ladybird -j1
[0/2] Re-checking globbed directories...
[65/2328] Building CXX object Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/boot/home/serenity2/AK/StackInfo.cpp.o
FAILED: Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/boot/home/serenity2/AK/StackInfo.cpp.o 
/bin/c++ -DENABLE_COMPILETIME_FORMAT_CHECK -DLibCore_EXPORTS -I/boot/home/serenity2 -I/boot/home/serenity2/Userland/Services -I/boot/home/serenity2/Userland/Libraries -I/boot/home/serenity2/Build/ladybird/Lagom -I/boot/home/serenity2/Build/ladybird/Lagom/Userland/Services -I/boot/home/serenity2/Build/ladybird/Lagom/Userland/Libraries -I/boot/home/serenity2/Meta/Lagom/../.. -I/boot/home/serenity2/Meta/Lagom/../../Userland -I/boot/home/serenity2/Meta/Lagom/../../Userland/Libraries -I/boot/home/serenity2/Meta/Lagom/../../Userland/Services -I/boot/home/serenity2/Build/ladybird -fPIC -Wall -Wextra -Wno-unknown-warning-option -Wno-unused-command-line-argument -fdiagnostics-color=always -fno-exceptions -Werror -Wno-expansion-to-defined -Wno-literal-suffix -Wno-dangling-reference -fno-semantic-interposition -fvisibility-inlines-hidden -Wno-maybe-uninitialized -Wno-shorten-64-to-32 -fsigned-char -g1 -O2 -fPIC -std=gnu++20 -MD -MT Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/boot/home/serenity2/AK/StackInfo.cpp.o -MF Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/boot/home/serenity2/AK/StackInfo.cpp.o.d -o Lagom/Userland/Libraries/LibCore/CMakeFiles/LibCore.dir/boot/home/serenity2/AK/StackInfo.cpp.o -c /boot/home/serenity2/AK/StackInfo.cpp
/boot/home/serenity2/AK/StackInfo.cpp: In constructor 'AK::StackInfo::StackInfo()':
/boot/home/serenity2/AK/StackInfo.cpp:41:15: error: 'pthread_getattr_np' was not declared in this scope; did you mean 'pthread_attr_t'?
   41 |     if ((rc = pthread_getattr_np(pthread_self(), &attr)) != 0) {
      |               ^~~~~~~~~~~~~~~~~~
      |               pthread_attr_t
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-shorten-64-to-32' may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option '-Wno-unused-command-line-argument' may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning-option' may have been intended to silence earlier diagnostics
ninja: build stopped: subcommand failed.
```
Thanks you very much for your fast improvements to Haiku, btw.
It's best if my workarounds are simply not needed.